Update docs related to React 18 (#35952)
vercel/next.js: docs/advanced-features/react-18/switchable-runtime.md

# Switchable Runtime (Alpha)

> By default, Next.js uses Node.js as the runtime for page rendering, including pre-rendering, server-side rendering.

Reviewer: ```suggestion
By default, Next.js uses Node.js as the runtime for page rendering, including pre-rendering and server-side rendering.
```

> If you have [React 18](/docs/advanced-features/react-18/overview) installed, there is a new experimental feature that lets you switch the page runtime between Node.js and [Edge Runtime](/docs/api-reference/edge-runtime). It affects [SSR streaming](/docs/advanced-features/react-18/streaming) and [Server Components](/docs/advanced-features/react-18/server-components) too.

Reviewer 2: ```suggestion
If you have [React 18](/docs/advanced-features/react-18/overview) installed, there is a new experimental feature that lets you switch the page runtime between Node.js and the [Edge Runtime](/docs/api-reference/edge-runtime). Changing the runtime affects [SSR streaming](/docs/advanced-features/react-18/streaming) and [Server Components](/docs/advanced-features/react-18/server-components) features, as well.
```

## Global Runtime Option

You can set the experimental option `runtime` to either `'nodejs'` or `'edge'` in your `next.config.js` file:

```jsx
// next.config.js
module.exports = {
  experimental: {
    runtime: 'nodejs',
  },
}
```

> This option determines which runtime should be used as the default page rendering runtime.

Reviewer 2: ```suggestion
This option determines which runtime should be used as the default rendering runtime for all pages.
```

> ## Per-page Runtime Option

Reviewer: ```suggestion
## Page Runtime Option
```

> On each page, it's also optional to export a `runtime` config set to either `'nodejs'` or `'edge'`:

Reviewer 2: ```suggestion
On each page, you can optionally export a `runtime` config set to either `'nodejs'` or `'edge'`:
```

```jsx
export const config = {
  runtime: 'nodejs',
}

> ```

Reviewer: Does this also work for `/pages/api/**`?

> When both are set, the per-page runtime option overrides the global runtime option. If the per-page runtime is not set, the global runtime option will be used.

Reviewer 2: ```suggestion
When both the per-page runtime and global runtime are set, the per-page runtime overrides the global runtime. If the per-page runtime is _not_ set, the global runtime option will be used.
```
